### PR TITLE
[SDK-2227] Add documentation around Browsers

### PR DESCRIPTION
**Pull request by frederikprijck**
This PR adds a small piece of documentation about changing the browser being used as well as how to implement a custom browser. Even though this isn't something the average user should need, adding a small amount of documentation around this can benefit those that have very specific needs.

### CODE REVIEW

File: docs-source/documentation/advanced-scenarios/browsers.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Browsers
+
+The SDK is configured to use a platform specific browser implementation to handle the redirect to Auth0. Depending on the platform, one or more Browser implementations might be available.
+An example is the Auth0.OidcClient.iOS SDK, which auto selects the browser based on the version of iOS used.
```
Comment (stevehobbsdev) [inline docs-source/documentation/advanced-scenarios/browsers.md:4]:
```suggestion
An example is the `Auth0.OidcClient.iOS` SDK, which auto-selects the browser based on the version of iOS used.
```

File: docs-source/documentation/advanced-scenarios/browsers.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Even though we do not recommend changing this, you could ensure all versions of iOS use the same browser by passing an explicit browser instance to the Browser property of the `Auth0ClientOptions`:
```
Comment (stevehobbsdev) [inline docs-source/documentation/advanced-scenarios/browsers.md:6]: Why is this not recommended? If there's a recommendation against doing this, we should explain why.

Or is this more of a "you will not normally need to do this"?

Comment (stevehobbsdev) [reply docs-source/documentation/advanced-scenarios/browsers.md:6]:
```suggestion
Even though we do not recommend changing this, you could ensure all versions of iOS use the same browser by passing an explicit browser instance to the `Browser` property of the `Auth0ClientOptions`:
```

File: docs-source/documentation/advanced-scenarios/browsers.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```
+var client = new Auth0Client(new Auth0ClientOptions {
+    Domain = "YOUR_AUTH0_DOMAIN",
+    ClientId = "YOUR_AUTH0_CLIENT_ID",
+    Browser = new SFSafariViewControllerBrowser()
+});
+```
+
+In case the default implementations that each SDK provides does not suit the needs for your application, each SDK allows to provide your own custom browser implementation by implementing the `IBrowser` interface.
```
Comment (stevehobbsdev) [inline docs-source/documentation/advanced-scenarios/browsers.md:16]:
```suggestion
In case the default implementations that each SDK provides does not suit the needs for your application, each SDK allows you to specify your own custom browser implementation by implementing the `IBrowser` interface.
```

File: docs-source/documentation/advanced-scenarios/browsers.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            ResultType = BrowserResultType.Success,
+            Response = callbackUrl
+        });
+
+        return tcs.Task;
+    }
+}
+```
+
+With the CustomBrowser implementation in place, instantiate `Auth0Client` with the appropriate `Auth0ClientOptions`:
```
Comment (stevehobbsdev) [inline docs-source/documentation/advanced-scenarios/browsers.md:40]:
```suggestion
With the `CustomBrowser` implementation in place, instantiate `Auth0Client` with the appropriate `Auth0ClientOptions`:
```

File: docs-source/documentation/advanced-scenarios/browsers.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+```
+var client = new Auth0Client(new Auth0ClientOptions {
+    Domain = "YOUR_AUTH0_DOMAIN",
+    ClientId = "YOUR_AUTH0_CLIENT_ID",
+    Browser = new CustomBrowser()
+});
+```
+
+The SDK will automatically call the Browser's Launch method and use the result that's being set as part of `tcs.SetResult()`.
```
Comment (stevehobbsdev) [inline docs-source/documentation/advanced-scenarios/browsers.md:50]: If "Browser" here is referring to the type, wrap it in backticks. Otherwise, should be lowercase "browser".

Backtick `Launch` as well if that's referring to an actual method.

File: docs-source/documentation/advanced-scenarios/browsers.md (hunk 2 above)
Comment (frederikprijck) [reply docs-source/documentation/advanced-scenarios/browsers.md:6]: > Why is this not recommended? If there's a recommendation against doing this, we should explain why.

Auto Select ensures it uses the best available option based on the iOS version. Changing that doesnt sounds like something to recommend. So I think it comes down to "You will not normally need this", let me rephrase that!

**Review by stevehobbsdev — APPROVED** [review]
